odl/discr/partition.py

```diff
@@ -0,0 +1,561 @@
+
+        Examples
+        --------
+        By default, the bounding box is inferred from the grid and
+        extends beyond the grid boundaries:
+
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid)  # Implicit begin and end
+        >>> part.bbox
+        Rectangle([-0.5, -1.5], [1.5, 3.0])
+
+        This behavior can be changed by explicitly giving begin and
+        end:
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.bbox
+        Rectangle([0.0, -1.0], [1.0, 2.0])
+        """
+        return self._bbox
+
+    @property
+    def begin(self):
+        """Vector with minimum bounding box coordinates."""
+        return self.bbox.begin
+
+    @property
+    def end(self):
+        """Vector with maximum bounding box coordinates."""
+        return self.bbox.end
+
+    def min(self):
+        """Return the 'minimum corner' of ``self.interv_prod``.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid)  # Implicit begin and end
+        >>> part.min()
+        array([-0.5, -1.5])
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.min()
+        array([ 0., -1.])
+
+        See also
+        --------
+        bbox
+        """
+        return self.bbox.min()
+
+    def max(self):
+        """Return the 'minimum corner' of ``self.interv_prod``.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid)  # Implicit begin and end
+        >>> part.max()
+        array([ 1.5,  3. ])
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.max()
+        array([ 1.,  2.])
+
+        See also
+        --------
+        bbox
+        """
+        return self.bbox.max()
+
+    def extent(self):
+        """Return a vector containing the total extent (max - min)."""
+        return self.max() - self.min()
+
+    def nodes_on_boundary(self):
+        """Return a list indicating if nodes lie on the boundary.
+
+        Returns
+        -------
+        on_bdry : `tuple` of 2-tuple of `bool`
+            Each 2-tuple contains the information whether the leftmost
+            (first entry) and rightmost (second entry) grid nodes lie
+            on the boundary of the bounding box in the corresponding
+            dimension.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid, begin=[0, -2], end=[2, 2])
+        >>> part.nodes_on_boundary()
+        ((True, False), (False, True))
+        """
+        bdry_list = []
+        for ax, (minpt, maxpt, bmin, bmax) in enumerate(zip(
+                self.grid.min_pt, self.grid.max_pt,
+                self.bbox.begin, self.bbox.end)):
+            bdry_list.append((minpt == bmin, maxpt == bmax))
```

Comment:
No so much in this case. The case "node on boundary" can only be reached by explicitly setting `begin` and/or `end`, so (maybe this should be documented) you would typically get `True` for, say, `begin` if you did
```python
>>> part = RectPartition(grid, begin=grid.min_pt)
```

I'll just have to think once more through the case where `begin` is totally arbitrary. Maybe it would be easier to have a switch like `extend_grid` or something where you could either choose to stop at the grid or extend it slightly as we did before (`as_midp`).